assets/js/laws-regs-parser.js

```diff
@@ -44,28 +44,98 @@ export default function parseLawsAndRegs(mainContent) {
       jumpLink.id = 'section' + i.toString();
       section.prepend(jumpLink);
     }
-    currentSubpart.appendChild(section);
+    const sectionWrapper = createDiv('section-wrapper');
+    sectionWrapper.appendChild(section);
+    currentSubpart.appendChild(sectionWrapper);
   });
 
   const newMainContent = buildMainContent(subparts);
 
   parent.insertBefore(newMainContent, mainContent);
   mainContent.remove();
-  const searchInput = document.querySelector('.searchbox');
-  searchInput.addEventListener('input', initSearch);
+  if (window.matchMedia('(max-width: 1023px)').matches) {
+    setUpMobileSearch();
+    return;
+  }
+  const searchBoxWrapper = document.querySelector('.desktop-search-box');
+  const searchInput = searchBoxWrapper.querySelector('.searchbox');
+  const searchGo = searchBoxWrapper.querySelector('.submit-search');
+  const searchNav = searchBoxWrapper.querySelector('.result-nav');
+  const searchBox = searchBoxWrapper.querySelector('.searchbox');
+  setUpButtons(searchGo, searchNav, searchBox);
+  searchInput.addEventListener('input', () =>
+    initSearch(searchInput, searchBoxWrapper, searchGo, searchNav)
+  );
+}
+
+function setUpMobileSearch() {
+  const searchBoxWrapper = document.querySelector('.mobile-search-box');
+  const searchInput = searchBoxWrapper.querySelector('.searchbox');
+  const searchGo = searchBoxWrapper.querySelector('.submit-search');
+  const searchNav = searchBoxWrapper.querySelector('.result-nav');
+  const searchBox = searchBoxWrapper.querySelector('.searchbox');
+  setUpButtons(searchGo, searchNav, searchBox);
+  searchInput.addEventListener('keydown', (e) => {
+    openOverlay(searchBoxWrapper, searchInput);
+  });
+  searchInput.addEventListener('click', () => {
+    openOverlay(searchBoxWrapper, searchInput);
+  });
+  searchInput.addEventListener('input', () =>
+    initSearch(searchInput, searchBoxWrapper, searchGo, searchNav)
+  );
 }
 
-function initSearch() {
-  const searchGo = document.querySelector('#submit-search');
-  const searchNav = document.querySelector('.result-nav');
+function openOverlay(searchBoxWrapper, searchInput) {
+  if (searchBoxWrapper.classList.contains('visible')) return;
+  searchBoxWrapper.classList.add('visible');
+  searchInput.focus();
+  let overlay = document.querySelector('.overlay');
+  if (overlay) {
+    overlay.classList.remove('display-none');
+    return;
+  }
+  overlay = document.createElement('div');
+  overlay.classList.add('overlay');
+  document.body.appendChild(overlay);
+  overlay.addEventListener('click', () => {
+    closeSearch(overlay, searchBoxWrapper, searchInput);
+  });
+}
+
+function initSearch(searchInput, searchBoxWrapper, searchGo, searchNav) {
   searchGo.classList.remove('display-none');
   searchNav.classList.add('display-none');
-  searchGo.addEventListener('click', search);
+  searchInput.addEventListener('keydown', (e) => {
+    if (e.key == 'Enter') {
+      search(searchBoxWrapper);
+    }
```

Comment:
Jumping between results is now quick, but I'm still seeing slow performance on this event. The chrome profiler seems to indicate that it's calling this code ten times:

<img width="489" alt="image" src="https://github.com/usdoj-crt/beta-ada/assets/15126660/63d84cd9-ac9d-47bd-8dfe-626c1d8e9985">

Ah - it looks like you're calling initSearch on input, which is going to add a keydown listener whenever you type into the box:

<img width="476" alt="image" src="https://github.com/usdoj-crt/beta-ada/assets/15126660/8699fc58-dd97-45e3-81ef-e0c6708ce1b6">

Yep - for this search, I typed `waffles the best I love them` then replaced it with `dog` and searched. The function is called once for every character in my search query:

<img width="647" alt="image" src="https://github.com/usdoj-crt/beta-ada/assets/15126660/9577d93a-a948-4576-bed8-e680e8f61ad9">